.github/workflows/ci.yml

```diff
@@ -12,34 +12,33 @@ jobs:
       fail-fast: false
       matrix:
         cfg:
-        - { ruby: 2.6, postgres: 12, gemfile: rails_6.0 }
-        - { ruby: 2.7, postgres: 12, gemfile: rails_6.0 }
-        - { ruby: 3.0.3, postgres: 12, gemfile: rails_6.0 }
-        - { ruby: 2.6, postgres: 13, gemfile: rails_6.0 }
-        - { ruby: 2.7, postgres: 13, gemfile: rails_6.0 }
-        - { ruby: 3.0.3, postgres: 13, gemfile: rails_6.0 }
-        - { ruby: 2.6, postgres: 12, gemfile: rails_6.1 }
-        - { ruby: 2.7, postgres: 12, gemfile: rails_6.1 }
-        - { ruby: 3.0.3, postgres: 12, gemfile: rails_6.1 }
-        - { ruby: 2.6, postgres: 13, gemfile: rails_6.1 }
-        - { ruby: 2.7, postgres: 13, gemfile: rails_6.1 }
-        - { ruby: 3.0.3, postgres: 13, gemfile: rails_6.1 }
-        - { ruby: 2.7, postgres: 12, gemfile: rails_7.0 }
-        - { ruby: 3.0.3, postgres: 12, gemfile: rails_7.0 }
-        - { ruby: 3.1, postgres: 12, gemfile: rails_7.0 }
-        - { ruby: 2.7, postgres: 13, gemfile: rails_7.0 }
-        - { ruby: 3.0.3, postgres: 13, gemfile: rails_7.0 }
-        - { ruby: 3.1, postgres: 13, gemfile: rails_7.0 }
-        - { ruby: 2.7, postgres: 12, gemfile: rails_7.1 }
-        - { ruby: 3.0.3, postgres: 12, gemfile: rails_7.1 }
-        - { ruby: 3.1, postgres: 12, gemfile: rails_7.1 }
-        - { ruby: 2.7, postgres: 13, gemfile: rails_7.1 }
-        - { ruby: 3.0.3, postgres: 13, gemfile: rails_7.1 }
-        - { ruby: 3.1, postgres: 13, gemfile: rails_7.1 }
-        - { ruby: 3.1, postgres: 12, gemfile: rails_7.2 }
-        - { ruby: 3.2, postgres: 12, gemfile: rails_7.2 }
-        - { ruby: 3.1, postgres: 13, gemfile: rails_7.2 }
-        - { ruby: 3.2, postgres: 13, gemfile: rails_7.2 }
+        - { ruby: 3.0.7, postgres: 14, gemfile: rails_6.0 }
+        - { ruby: 3.0.7, postgres: 15, gemfile: rails_6.0 }
```

Comment:
seems like postgres 15+ will explodes on dump version mismatch 🤔

Reply:
`pg_dump` does _not_ want to be working with a server version from the future -- the other way around is mostly fine.

You could use `ubuntu-22.04` for a `runs_on` target which installs v16 by default (this will relatively soon be the target for `ubuntu-latest` — see https://github.com/actions/runner-images/issues/10636)

Alternatively, [manually installing](https://wiki.postgresql.org/wiki/Apt#Manual_Repository_Configuration) an updated version of the PostgreSQL client libraries would help:

```
- name: Update postgres
  run: |
     sudo sh -c 'echo "deb [signed-by=/usr/share/postgresql-common/pgdg/apt.postgresql.org.asc] https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
     sudo apt-get update && sudo apt-get install postgresql-client-16
```